traffic_ops/traffic_ops_golang/federation_resolvers/federation_resolvers.go

```diff
@@ -136,3 +151,99 @@ func Read(w http.ResponseWriter, r *http.Request) {
 
 	api.WriteResp(w, r, resolvers)
 }
+
+func Delete(w http.ResponseWriter, r *http.Request) {
+	inf, sysErr, userErr, errCode := api.NewInfo(r, []string{"id"}, []string{"id"})
+	tx := inf.Tx.Tx
+	if sysErr != nil || userErr != nil {
+		api.HandleErr(w, r, tx, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
+
+	userErr, sysErr, statusCode, alert, respObj := delete(inf)
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, tx, statusCode, userErr, sysErr)
+		return
+	}
+
+	api.WriteRespAlertObj(w, r, tc.SuccessLevel, alert.Text, respObj)
+}
+
+func delete(inf *api.APIInfo) (error, error, int, tc.Alert, tc.FederationResolver) {
+	var userErr error
+	var sysErr error
+	var statusCode = http.StatusOK
+	var alert tc.Alert
+	var result tc.FederationResolver
+
+	err := inf.Tx.Tx.QueryRow(deleteQuery, inf.IntParams["id"]).Scan(&result.ID, &result.IPAddress, &result.Type)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			userErr = fmt.Errorf("No federation resolver by ID %d", inf.IntParams["id"])
+			statusCode = http.StatusNotFound
+		} else {
+			userErr, sysErr, statusCode = api.ParseDBError(err)
+		}
+		return userErr, sysErr, statusCode, alert, result
+	}
+
+	changeLogMsg := fmt.Sprintf("FEDERATION_RESOLVER: %s, ID: %d, ACTION: Deleted", *result.IPAddress, *result.ID)
+	api.CreateChangeLogRawTx(api.ApiChange, changeLogMsg, inf.User, inf.Tx.Tx)
```

Comment:
A while ago someone went through the TO Go codebase and made all of our changelogs conform to a specific format - which is what I have there. I think it would've been much better to edit the now-unused `api.CreateChangeLogBuildMsg` but such is life.

Reply:
Are you saying that `api.CreateChangeLogBuildMsg` produces the wrong format? If so, should it be updated or deleted? Also, if it were updated then all code using it would also be fixed, including this code here if it were using it. It seems to me that using separate calls to `fmt.Sprintf` all over the code is a poor way to ensure the formatting these are consistent.

That said, it is just a code quality suggestion, not necessarily a correctness issue, so do as you think best.

Reply:
> Are you saying that `api.CreateChangeLogBuildMsg` produces the wrong format? If so, should it be updated or deleted?

Yes and yes. Preferably updated.

> It seems to me that using separate calls to fmt.Sprintf all over the code is a poor way to ensure the formatting these are consistent.

It is and that ought to be changed. I'd just rather do it in a separate PR when/if I find the time.